### ISSUE TITLE
Inventory plugin broken with ansible-core

### ISSUE THREAD

**Issue by cmammoli** [netbox-community/ansible_modules#490]
##### ISSUE TYPE
 - Bug Report

##### SOFTWARE VERSIONS
ansible [core 2.11.0rc2]

###### Ansible:
ansible [core 2.11.0rc2]

###### Netbox:
netbox 2.10.9 but it doesn't matter, a connection to the api doesn't happen

###### Collection:
netbox-netbox: 3.0.0

##### SUMMARY
Trying to use the inventory plugin with ansible core causes the following error:
```
[WARNING]:  * Failed to parse /home/c.mammoli/devel/ansible/inventory_netbox.yml with ansible_collections.netbox.netbox.plugins.inventory.nb_inventory
plugin: Invalid value "['tag', 'site', 'tenant', 'role', 'cluster', 'rack', 'platform', 'is_virtual', 'rack', 'rack_group']" for configuration option
"plugin_type: inventory plugin: ansible_collections.netbox.netbox.plugins.inventory.nb_inventory setting: group_by ", valid values are: ['sites', 'site',
'tenants', 'tenant', 'racks', 'rack', 'rack_group', 'rack_role', 'tags', 'tag', 'device_roles', 'role', 'device_types', 'device_type', 'manufacturers',
'manufacturer', 'platforms', 'platform', 'region', 'cluster', 'cluster_type', 'cluster_group', 'is_virtual', 'services', 'status']
```

The value of group_by does not matter, it can be empty or set to any value

##### STEPS TO REPRODUCE
pip install ansible-core
ansible-galaxy collection install netbox.netbox
Create a simple inventory config file for netbox, mine is:

```
plugin: netbox.netbox.nb_inventory
validate_certs: True
token: foo
api_endpoint : bar
config_context: True
plurals: false
flatten_config_context: True
flatten_local_context_data: True
group_by:
  - tag
  - site
  - tenant
  - role
  - cluster
  - rack
  - platform
  - is_virtual
  - rack
  - rack_group
query_filters:
  - tag__n: 'no_automation'
  - has_primary_ip: 'true'
  - status: 'active'
```
But any config won't work, even the example in the module documentation

Then run
ansible-inventory -i myinventory.yml --list

##### EXPECTED RESULTS
The inventory is printed

##### ACTUAL RESULTS

```
c.mammoli ~/devel/ansible $ ansible-inventory -i inventory_netbox.yml  --list
[DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the controller starting with Ansible 2.12. Current version: 3.7.3 (default, Jul 25 2020,
13:03:44) [GCC 8.3.0]. This feature will be removed from ansible-core in version 2.12. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[WARNING]: Failed to load inventory plugin, skipping tower
[WARNING]:  * Failed to parse /mnt/c/Users/c.mammoli/devel/ansible/inventory_netbox.yml with yaml plugin: Plugin configuration YAML file, not YAML inventory
[WARNING]:  * Failed to parse /mnt/c/Users/c.mammoli/devel/ansible/inventory_netbox.yml with ini plugin: Invalid host pattern 'plugin:' supplied, ending in
':' is not allowed, this character is reserved to provide a port.
[WARNING]:  * Failed to parse /mnt/c/Users/c.mammoli/devel/ansible/inventory_netbox.yml with ansible_collections.netbox.netbox.plugins.inventory.nb_inventory
plugin: Invalid value "['tag', 'site', 'tenant', 'role', 'cluster', 'rack', 'platform', 'is_virtual', 'rack', 'rack_group']" for configuration option
"plugin_type: inventory plugin: ansible_collections.netbox.netbox.plugins.inventory.nb_inventory setting: group_by ", valid values are: ['sites', 'site',
'tenants', 'tenant', 'racks', 'rack', 'rack_group', 'rack_role', 'tags', 'tag', 'device_roles', 'role', 'device_types', 'device_type', 'manufacturers',
'manufacturer', 'platforms', 'platform', 'region', 'cluster', 'cluster_type', 'cluster_group', 'is_virtual', 'services', 'status']
[WARNING]: Unable to parse /mnt/c/Users/c.mammoli/devel/ansible/inventory_netbox.yml as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
{
    "_meta": {
        "hostvars": {}
    },
    "all": {
        "children": [
            "ungrouped"
        ]
    }
}

```

**Comment by jvanderaa** [comment]
Do other inventories work? If not, this may be an issue in the RC rather than the current release of NetBox.netbox. I would open an issue on Ansible for this.

**Comment by knorring** [comment]
This could be related: https://github.com/netbox-community/ansible_modules/issues/474#issuecomment-819541856

**Comment by diekgait** [comment]
I see exactly the same issue after upgrading Netbox from 2.10.6 to 2.11.0. Using pynetbox directly still works fine.

**Comment by diekgait** [comment]
@cmammoli @jvanderaa This issue is caused by the rename of "rack-groups" into "locations" in Netbox version 2.11.x please check Breaking changes in the release notes:
https://github.com/netbox-community/netbox/releases

In nb_inventory.py on line 814 change:
"/api/dcim/rack-groups/?limit=0"
into:
"/api/dcim/locations/?limit=0"
And it will work again. Since the entire module has been renamed it may be elegant to replace "rack-groups" with "locations" in other places as well.

**Comment by cmammoli** [comment]
@diekgait nope, I'm testing against netbox 2.10.9, the only thing I changed was the ansible version

**Comment by diekgait** [comment]
@cmammoli  You are right, I didn't read your initial post good enough. I will create a new issue for this.

**Comment by knorring** [comment]
@cmammoli, did you check out https://github.com/netbox-community/ansible_modules/issues/474#issuecomment-819541856? The `group_by` issue is fixed in the development branch of Ansible (see https://github.com/ansible/ansible/issues/74225).

A quick fix that might help is described in https://github.com/netbox-community/ansible_modules/issues/474#issuecomment-819551572.

Hope it helps.

**Comment by FragmentedPacket** [comment]
@cmammoli Did you get your issue resolved with the help of everyone?

**Comment by FlxPeters** [comment]
I have the exact same problem with Netbox v2.9.9 and `quay.io/ansible/awx-ee:0.2.0` as execution environment in AWX. 
Also tried older versions of `quay.io/ansible/awx-ee` but they all contain ansible-core which seams to be the root of the problem. 

The quickfix mentioned here does not help because AWX will pull and use the file from Galaxy.

**Comment by FlxPeters** [comment]
My current workaround is to use a forked version of the collection with the mentioned quickfix applied.

```
---
collections:
  - name: https://github.com/FlxPeters/ansible_modules.git
    type: git
    version: devel
``` 

This is not a stable fix, but does the job for now until https://github.com/ansible/ansible/pull/74267 is available in any `quay.io/ansible/awx-ee` image. 

A possible solution would be to remove the choices in the inventory plugin for know. Should i open an PR for this?

**Comment by jvanderaa** [comment]
This fixed now that 2.11.1 core is out?

**Comment by ryanmerolle** [comment]
It works for me using https://demo.netbox.dev which at this posting is running 2.11.3.  If you want to try, you will need to readd the below token since the demo gets reset every day.

inventory.yml (I commented out items based on the version of netbox and the demo data)

```
plugin: netbox.netbox.nb_inventory
validate_certs: True
token: 157b39a44816457929721617cbca0ef6bd55d525
api_endpoint: https://demo.netbox.dev
config_context: True
plurals: false
flatten_config_context: True
flatten_local_context_data: True
group_by:
  - tag
  - site
  - tenant
  - role
  - cluster
  - rack
  - platform
  - is_virtual
  - rack
    #- rack_group
query_filters:
        # - tag__n: 'no_automation'
        # - has_primary_ip: 'true'
  - status: 'active'
```
output:
```
$ ansible-inventory -i inventory.yml --list
{
    "_meta": {
        "hostvars": {
            "07bc873d-0920-419b-8bd6-c17b5c544bd6": {
                "custom_fields": {
                    "Solarwinds URL": null
                },
                "device_type": "48-port-patch-panel",
                "is_virtual": false,
                "locations": [],
                "manufacturer": "panduit",
                "rack": "Comms closet",
                "regions": [
                    "us-ny",
                    "us",
                    "north-america"
                ],
                "role": "patch-panel",
                "services": [],
                "site": "dm-albany",
                "status": {
                    "label": "Active",
                    "value": "active"
                },
                "tags": []
            },
            "0caa9947-fe07-434a-bfde-fdab0fcf3c94": {
                "custom_fields": {
                    "Solarwinds URL": null
                },
                "device_type": "48-port-patch-panel",
                "is_virtual": false,
                "locations": [],
                "manufacturer": "panduit",
                "rack": "Comms closet",
                "regions": [
                    "us-oh",
                    "us",
                    "north-america"
                ],
                "role": "patch-panel",
                "services": [],
                "site": "dm-akron",
                "status": {
                    "label": "Active",
                    "value": "active"
[ommitted...]
```
